aws/sdk/examples/sts/src/bin/credentials-provider.rs

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_auth::{CredentialsError, ProvideCredentials};
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, SystemTime};
+use sts::Credentials;
+
+#[tokio::main]
+async fn main() -> Result<(), dynamodb::Error> {
+    tracing_subscriber::fmt::init();
+    let client = sts::Client::from_env();
+    let sts_provider = StsCredentialsProvider {
+        client,
+        credentials: Arc::new(Mutex::new(None)),
+    };
+    sts_provider.spawn_refresh_loop().await;
```

Comment:
Is the Tokio runtime smart enough to discontinue the refresh loop after `main` completes?

Reply:
yep

Reply:
the runtime is dropped which cancels all tasks